### PR TITLE
Handle nested types

### PR DESCRIPTION
**Pull request by Aniket21mathur**
Src -> https://github.com/Aniket21mathur/ProtocolBuffers-Examples/blob/master/chapel/src/nestedTypes.proto
Build -> https://github.com/Aniket21mathur/ProtocolBuffers-Examples/blob/master/chapel/build/nestedTypes.chpl

### CODE REVIEW

**Comment by Aniket21mathur** [comment]
This pull request should be rebased with master once #22 is merged. Only 4cedbc1 is related to this pr.

File: src/plugin/reflection_class.cpp [hunk 1]
```diff
@@ -65,8 +65,33 @@ namespace chapel {
       }
     }
 
+    // write grand chindren: Message nested types
```
Comment (lydia-duncan) [inline src/plugin/reflection_class.cpp]: Typo on "chindren" :p

File: src/plugin/reflection_class.cpp [hunk 2]
```diff
@@ -65,8 +65,33 @@ namespace chapel {
...
+    // handled here since nested records and enum declaration in records
+    // is not yet supported in Chapel.
```
Comment (lydia-duncan) [inline src/plugin/reflection_class.cpp]: "is not" -> "are not", you are talking about two categories so it should be plural

File: test/endToEnd/enums/protoFile/enums.proto [hunk 3]
```diff
@@ -10,4 +10,10 @@ message enumTest {
   color a = 1; 
   int64 b = 2;
   repeated color c = 3;
+  fruit d = 4;
```
Comment (lydia-duncan) [inline test/endToEnd/enums/protoFile/enums.proto:13]: We should test repeated instances of nested types as well

**Review by lydia-duncan — COMMENTED** [review]
Couple of comment clean ups and a request for additional testing

File: src/plugin/reflection_class.cpp (hunk 1 above)
Comment (Aniket21mathur) [reply src/plugin/reflection_class.cpp]: Opps! Dumb me xD

File: src/plugin/reflection_class.cpp (hunk 2 above)
Comment (Aniket21mathur) [reply src/plugin/reflection_class.cpp]: Yes, I will correct it :sweat_smile:

File: test/endToEnd/enums/protoFile/enums.proto (hunk 3 above)
Comment (Aniket21mathur) [reply test/endToEnd/enums/protoFile/enums.proto:13]: Good point! I will add tests for that :)

**Comment by Aniket21mathur** [comment]
@lydia-duncan made the changes in 1090fe8 :)

**Review by lydia-duncan — APPROVED** [review]
Looks good, thanks!

**Pull request merged**